src/controllers/token.rs

```diff
@@ -114,3 +117,133 @@ pub fn revoke_current(req: &mut dyn RequestExt) -> EndpointResult {
 
     Ok(Response::builder().status(204).body(Body::empty()).unwrap())
 }
+
+/// Verifies that the GitHub signature in request headers is valid
+fn verify_github_signature(req: &dyn RequestExt, json: &[u8]) -> Result<(), Box<dyn AppError>> {
+    // Read and decode request headers
+    let headers = req.headers();
+    let req_key_id = headers
+        .get("GITHUB-PUBLIC-KEY-IDENTIFIER")
+        .ok_or_else(|| bad_request("missing HTTP header: GITHUB-PUBLIC-KEY-IDENTIFIER"))?
+        .to_str()
+        .map_err(|e| bad_request(&format!("failed to decode HTTP header: {e:?}")))?;
+    let sig = headers
+        .get("GITHUB-PUBLIC-KEY-SIGNATURE")
+        .ok_or_else(|| bad_request("missing HTTP header: GITHUB-PUBLIC-KEY-SIGNATURE"))?;
+    let sig = base64::decode(sig)
+        .map_err(|e| bad_request(&format!("failed to decode signature as base64: {e:?}")))?;
+
+    // Fetch list of public keys from GitHub API
+    let app = req.app();
+    let public_keys = app
+        .github
+        .public_keys(&app.config.gh_client_id, &app.config.gh_client_secret)
+        .map_err(|e| bad_request(&format!("failed to fetch GitHub public keys: {e:?}")))?;
```

Comment:
do you think it would make sense to cache the keys to avoid hitting the GitHub API rate limit?

Reply:
I'm happy to add caching if you would prefer. I couldn't see any other responses from GitHub being cached, so I assumed this would be OK too.

Reply:
:wave: 
We recommend making authenticated requests (e.g., with a PAT) to fetch the public keys to avoid rate limiting.

Reply:
We do make an authenticated request to fetch the public keys but that counts towards the 5000 requests per hour rate limit.

Reply:
I think you'd be fairly safe caching these for a very long time (like for the lifetime of the process), assuming you accounted for if we started signing with a new key, you'd re-fetch.

Reply:
I added a cache for downloaded public keys that is valid for 24 hours, which seemed like a reasonable balance to strike.